Makefile

```diff
@@ -0,0 +1,59 @@
+ENV_TYPE ?= venv
+PYTHON_VERSION ?= 3.10
+VENV_DIR := .venv
+CONDA_ENV ?= onemod
+
+.PHONY: list-vars check-python setup clean mypy pre-commit
+
+list-vars:  ## List variables
+	@echo "ENV_TYPE        = $(ENV_TYPE)"
+	@echo "PYTHON_VERSION  = $(PYTHON_VERSION)"
+	@echo "VENV_DIR        = $(VENV_DIR)"
+	@echo "CONDA_ENV       = $(CONDA_ENV)"
+
+check-python:  ## Check if the specified Python version is installed (venv only)
+	@if [ "$(ENV_TYPE)" = "venv" ]; then \
+		if ! command -v python$(PYTHON_VERSION) >/dev/null 2>&1; then \
+			echo "Error: Python $(PYTHON_VERSION) is not installed."; \
+			exit 1; \
+		fi; \
+	fi
+
+setup: check-python  ## Set up the development environment (venv or conda)
+ifeq ($(ENV_TYPE),venv)
+	python$(PYTHON_VERSION) -m venv $(VENV_DIR)
+	$(VENV_DIR)/bin/pip install -U pip
+	$(VENV_DIR)/bin/pip install -e ".[dev]"
```

Comment:
Could we have a way to install other optional dependencies with the make file (e.g., jobmon, docs...)?

Reply:
Yes. Not sure whether that's additional options we need to add to the Makefile. I don't mind it, but personally I'd probably just install manually with `pip install -e ".[docs]"`, etc. after environment setup

Reply:
Okay, it's fine to keep things simple